repo-agent/repowatch/controllers/repowatch_controller.go

```diff
@@ -54,6 +56,55 @@ var seededRand = rand.New(
 
 type githubClientFactory func(ctx context.Context, k8sClient client.Client, repoWatch *reviewv1alpha1.RepoWatch) (*github.Client, map[string]string, error)
 
+// PersistingTokenSource wraps an oauth2.TokenSource and persists the token to a Kubernetes secret when it changes.
+type PersistingTokenSource struct {
+	Source     oauth2.TokenSource
+	K8sClient  client.Client
+	SecretName string
+	Namespace  string
+}
+
+func (s *PersistingTokenSource) Token() (*oauth2.Token, error) {
+	t, err := s.Source.Token()
+	if err != nil {
+		return nil, err
+	}
+
+	// Persist the token if it has changed
+	go func(token *oauth2.Token) {
+		ctx := context.Background()
```

Comment:
Starting a go routine here without any mechanism to signal cancellation (e.g., from the parent context) can lead to orphaned goroutines, especially under high load or during pod termination. The background context `context.Background()` does not carry deadlines or cancellation signals. Consider using a context derived from the reconciler's context to ensure the update operation can be cancelled if the reconciliation is cancelled.

Reply:
Done